packages/cosmos_ui_components/lib/components/cosmos_switch.dart

```diff
@@ -13,26 +14,89 @@ class CosmosSwitch extends StatelessWidget {
+      ..add(DiagnosticsProperty<bool>('checked', checked));
+  }
+}
+
+class CosmosSwitchState extends State<CosmosSwitch> with SingleTickerProviderStateMixin {
+  late Animation<Alignment> _circleAnimation;
+  late AnimationController _animationController;
+
+  @override
+  void initState() {
+    super.initState();
+    _animationController = AnimationController(vsync: this, duration: const Duration(milliseconds: 200));
+    _circleAnimation = AlignmentTween(
+      begin: widget.checked ? Alignment.centerRight : Alignment.centerLeft,
+      end: widget.checked ? Alignment.centerLeft : Alignment.centerRight,
+    ).animate(
+      CurvedAnimation(
+        parent: _animationController,
+        curve: Curves.linear,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTap: () {
+        if (_animationController.isCompleted) {
+          _animationController.reverse();
+        } else {
+          _animationController.forward();
+        }
+        widget.onChanged!(!widget.checked);
+      },
+      child: _AnimatedSwitch(animation: _circleAnimation, checked: widget.checked),
+    );
+  }
+}
+
+class _AnimatedSwitch extends AnimatedWidget {
+  const _AnimatedSwitch({
+    required Animation<Alignment> animation,
+    required this.checked,
+    Key? key,
+  }) : super(key: key, listenable: animation);
+
+  final bool checked;
+
   @override
   Widget build(BuildContext context) {
-    final theme = CosmosTheme.of(context);
+    final animation = listenable as Animation<Alignment>;
     return Transform.scale(
-      scale: 0.9, // so that the border appears closer to the switch
+      scale: 0.9,
       child: Container(
+        width: 50,
+        height: 28,
         decoration: BoxDecoration(
-          borderRadius: BorderRadius.circular(1000),
-          border: Border.all(
-            color: checked ? Colors.transparent : theme.colors.text.withOpacity(0.1),
-          ),
+          borderRadius: BorderRadius.circular(30),
+          gradient: checked
+              ? const LinearGradient(
+                  stops: [0, 0.5866, 1],
+                  colors: GlobalConstants.cosmosGradientColors,
+                )
+              : null,
+          color: checked ? null : CosmosTheme.of(context).colors.text,
         ),
         child: Transform.scale(
-          scale: 1.1, // so that the border appears closer to the switch
-          child: CupertinoSwitch(
-            value: checked,
-            onChanged: onChanged,
-            activeColor: theme.colors.link,
-            thumbColor: checked ? theme.colors.background : theme.colors.text,
-            trackColor: theme.colors.background,
+          scale: 1.1,
+          child: Align(
+            alignment: animation.value,
+            child: SizedBox(
+              width: 30,
+              height: 30,
+              child: Image.asset('assets/images/toggle.png'),
```

Comment:
let's not hardcode the asset here, since this component is reused in both starport template and emeris (and potentially other apps) we might want to customize the toggle
```suggestion
              child: Image.asset(toggleAsset),
```